docs/inter-domain-integration/metering-point-period-schema.py

```diff
@@ -0,0 +1,116 @@
+All but the `ToDate` of the last period must have value. The `ToDate` of the last period is null.
+
+Data must be stored in a Delta table.
+The table holds all consumption, production, exchange, and child metering points.
+The table must be partitioned by `ToDate`: ToDate_Year/ToDate_Month/ToDate_Date.
+TODO explain why by to-date
+The table data must always contain updated periods.
+"""
+metering_point_period_schema = StructType(
+    [
+        # GSRN (18 characters) that uniquely identifies the metering point
+        # Used in balance fixing and settlement.
+        # Example: 578710000000000103
+        StructField("MeteringPointId", StringType(), False),
+        
+        # "E17" (consumption) | "E18" (production) | "E20" (exchange) | "D01"-"D99" (child)
+        # Used in balance fixing and settlement. However, child metering points are only used in settlement.
+        # Example: E20
+        StructField("Type", StringType(), False),
+        
+        # "E02" (quarterly)| "D01" (flex)
+        # Used in balance fixing and settlement.
+        # Example: D01
+        StructField("SettlementMethod", StringType(), False),
+        
+        # 3 character grid area code uniquely identifying the grid area. All characters must be digits (0-9).
+        # Used in balance fixing and settlement.
+        # Example: 805
+        StructField("GridAreaCode", StringType(), False),
+        
+        # "E22" (connected) | "E23" (disconnected) | "D03" (new) | "D02" (closed down)
+        # Used in balance fixing and settlement.
+        # Example: E22
+        StructField("ConnectionState", StringType(), False),
+        
+        # "PT1H" (hourly) | "PT15M" (quarterly)
+        # Used in balance fixing and settlement.
+        # Example: PT1H
+        StructField("Resolution", StringType(), False),
+        
+        # 3 character grid area code uniquely identifying the from-grid area. All characters must be digits (0-9).
+        # The code has a value for E20 (exchange) metering points. For all other types it's null.
+        # Used in balance fixing and settlement.
+        # Example: 122
+        StructField("FromGridAreaCode", StringType(), True),
+        
+        # 3 character grid area code uniquely identifying the to-grid area. All characters must be digits (0-9).
+        # The code has a value for E20 (exchange) metering points. For all other types it's null.
+        # Used in balance fixing and settlement.
+        # Example: 134
+        StructField("ToGridAreaCode", StringType(), True),
+        
+        # The id of the parent metering point or null.
+        # Used only in settlement.
+        # Example: 578710000000000112
+        StructField("ParentMeteringPointId", StringType(), True),
+        
+        # The unique GLN/EIC number of the energy supplier
+        # Used in balance fixing and settlement.
+        # Example: 8100000000108
+        StructField("EnergySupplierId", StringType(), False),
```

Comment:
Not all MPs have a supplier registered, e.g. E20
```suggestion
        StructField("EnergySupplierId", StringType(), True),
```

Reply:
Thx, fixed and doc added.